frontend/components/Filter/FilterBanner.tsx

```diff
@@ -11,90 +11,353 @@ import RugbyImage from "./../../public/images/rugby-ball.png";
+    }
+
+    const [selectedSport, setSelectedSport] = useState("");
+
+    const icons = {
+        volleyball: {
+            image: VolleyballImage,
+            style: "w-8 h-8",
+            sport_name: VOLLEYBALL_SPORT_STRING,
+        },
+        badminton: {
+            image: BadmintonImage,
+            style: "w-8 h-8",
+            sport_name: BADMINTON_SPORT_STRING,
+        },
+        basketball: {
+            image: BasketballImage,
+            style: "w-8 h-8",
+            sport_name: BASKETBALL_SPORT_STRING,
+        },
+        soccer: {
+            image: SoccerImage,
+            style: "w-8 h-8",
+            sport_name: SOCCER_SPORT_STRING,
+        },
+        tennis: {
+            image: TennisImage,
+            style: "w-8 h-8",
+            sport_name: TENNIS_SPORT_STRING,
+        },
+        "table tennis": {
+            image: PingPongImage,
+            style: "w-8 h-8",
+            sport_name: TABLE_TENNIS_SPORT_STRING,
+        },
+        oztag: {
+            image: RugbyImage,
+            style: "w-8 h-8",
+            sport_name: OZTAG_SPORT_STRING,
+        },
+        baseball: {
+            image: BaseballImage,
+            style: "w-8 h-8",
+            sport_name: BASEBALL_SPORT_STRING,
+        },
+    };
+
+    const scroll = () => {
+        document.getElementById("filter-overflow")?.scrollBy({
+            top: 0,
+            left: 50,
+            behavior: "smooth",
+        });
+    };
+
+    async function applyFilters(selectedSportProp: string) {
+        const isAnyPriceBool = maxPriceSliderValue === PRICE_SLIDER_MAX_VALUE;
+        // changed it so that instead of it running only if its not max, if locaiton is not ""
+        const isAnyProximityBool =
+            srcLocation !== "" ||
+            maxProximitySliderValue === PROXIMITY_SLIDER_MAX_VALUE;
+
+        let filteredEventDataList = [...eventDataList];
+        console.log(srcLocation);
+        // Filter by MAX PRICE
+        if (!isAnyPriceBool) {
+            let newEventDataList = filterEventsByPrice(
+                [...filteredEventDataList],
+                null,
+                maxPriceSliderValue
+            );
+            filteredEventDataList = newEventDataList;
+        }
+        setAppliedMaxPriceSliderValue(maxPriceSliderValue);
+
+        // Filter by DATERANGE
+        if (dateRange.startDate && dateRange.endDate) {
+            let newEventDataList = filterEventsByDate(
+                [...filteredEventDataList],
+                Timestamp.fromDate(
+                    new Date(dateRange.startDate + DAY_START_TIME_STRING)
+                ), // TODO: needed to specify maximum time range on particular day.
+                Timestamp.fromDate(
+                    new Date(dateRange.endDate + DAY_END_TIME_STRING)
+                )
+            );
+            filteredEventDataList = newEventDataList;
+        }
+        setAppliedDateRange(dateRange);
+
+        // Filter by MAX PROXIMITY
```

Comment:
search bar currently does not filter by proximity properly.
e.g., searching up Perth WA Australia within 50km does not filter out all events